anise/src/almanac/mod.rs

```diff
@@ -164,9 +164,27 @@ impl Almanac {
         }
 
         if let Ok(metadata) = Metadata::decode_header(&bytes) {
+            // Use `try_from` to validate the dataset type
+            let dataset_type =
+                DataSetType::try_from(metadata.dataset_type as u8).map_err(|err| {
+                    AlmanacError::GenericError {
+                        err: format!("Invalid dataset type: {}", err),
+                    }
+                })?;
+
             // Now, we can load this depending on the kind of data that it is
-            match metadata.dataset_type {
-                DataSetType::NotApplicable => unreachable!("no such ANISE data yet"),
+            match dataset_type {
+                DataSetType::NotApplicable => {
+                    // Not something that can be decoded
+                    warn!(
+                        "Encountered unsupported dataset type: {:?} in {}",
```

Comment:
```suggestion
                        "Malformed dataset type in {}",
```

If it's decoded to be `00` (which is the only time we'd see the `NotApplicable` enum variant), then the original encoding likely failed, or the bytes could not be read from disk/ram. Therefore, I think it's reasonable to omit the `NotApplicable` in the message, and just tell the user something is wrong as you've done.

Reply:
Based on the comment below, I don't believe this is relevant any longer. Will leave open incase https://github.com/nyx-space/anise/pull/418/commits/601854543968ee1a27a268491aef70359baab227 wasn't updated in the way you were thinking.